packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -317,13 +317,23 @@ class Auth {
         const refreshToken = refreshTokenRegistered || refreshTokenGuest
         if (refreshToken) {
             try {
-                return this.queueRequest(
+                return await this.queueRequest(
                     () => helpers.refreshAccessToken(this.client, {refreshToken}),
                     !!refreshTokenGuest
                 )
-            } catch {
-                // If anything bad happens during refresh token flow
-                // we continue with the PKCE guest user flow.
+            } catch (error) {
+                // If the refresh token is invalid, we need to re-login the user
+                if (error instanceof Error && 'response' in error) {
+                    // commerce-sdk-isomorphic throws a `ResponseError`, but doesn't export the class.
+                    // We can't use `instanceof`, so instead we just check for the `response` property
+                    // and assume it is a fetch Response.
+                    const json = await (error['response'] as Response).json()
```

Comment:
Just want to call out that the bracket access is because `.response` was causing a type error. We didn't get to the bottom of it, but IIRC there's a tsconfig setting that controls whether unknown properties on a type can be accessed using dot notation, or if they must use brackets to visibly indicate "this property _might_ exist, but maybe not". This feels related to that, although we didn't investigate long enough to verify.